airflow/utils/log/logging_mixin.py

```diff
@@ -46,9 +46,10 @@ def logger(self):
         return self.log
 
     @property
-    def log(self):
+    def log(self) -> Logger:
         try:
-            return self._log
+            # FIXME: LoggingMixin should have a default _log field.
```

Comment:
This comment doesn't make sense - the default is handed 2 lines down.

Reply:
The concern is that mypy can not infer the type of self._log if self._log is not set in __init__, so we need a 'ignore mypy' comment.

Reply:
![image](https://user-images.githubusercontent.com/8811558/62885204-277b9080-bd30-11e9-9ae7-e6efc16bbc9e.png)

This is what @coufon means

Reply:
Ah, we should replace this with 

```python
    @cached_property
    def log(self):
        return logging.root.getChild(
            self.__class__.__module__ + '.' + self.__class__.__name__
        )
```

Though do we have to change this at all in this PR? It's a change that seems unrelated to the topic of the PR.

Reply:
Nope, we can ignore it.